src/ctap/pin_protocol_v1.rs

```diff
@@ -0,0 +1,1137 @@
+        // User verifications is only supported through PIN currently.
+        #[cfg(not(feature = "with_ctap2_1"))]
+        {
+            Err(Ctap2StatusCode::CTAP1_ERR_INVALID_COMMAND)
+        }
+        #[cfg(feature = "with_ctap2_1")]
+        {
+            Err(Ctap2StatusCode::CTAP2_ERR_INVALID_SUBCOMMAND)
+        }
+    }
+
+    #[cfg(feature = "with_ctap2_1")]
+    fn process_get_uv_retries(&self) -> Result<AuthenticatorClientPinResponse, Ctap2StatusCode> {
+        // User verifications is only supported through PIN currently.
+        #[cfg(not(feature = "with_ctap2_1"))]
+        {
+            Err(Ctap2StatusCode::CTAP1_ERR_INVALID_COMMAND)
+        }
+        #[cfg(feature = "with_ctap2_1")]
+        {
+            Err(Ctap2StatusCode::CTAP2_ERR_INVALID_SUBCOMMAND)
+        }
+    }
+
+    #[cfg(feature = "with_ctap2_1")]
+    fn process_set_min_pin_length(
+        &mut self,
+        persistent_store: &mut PersistentStore,
+        min_pin_length: u8,
+        min_pin_length_rp_ids: Option<Vec<String>>,
+        pin_auth: Option<Vec<u8>>,
+    ) -> Result<(), Ctap2StatusCode> {
+        if min_pin_length_rp_ids.is_some() {
+            return Err(Ctap2StatusCode::CTAP2_ERR_UNSUPPORTED_EXTENSION);
+        }
+        if persistent_store.pin_hash().is_some() {
+            match pin_auth {
+                Some(pin_auth) => {
+                    // TODO(kaczmarczyck) not mentioned, but maybe useful?
+                    // if persistent_store.pin_retries() == 0 {
+                    //     return Err(Ctap2StatusCode::CTAP2_ERR_PIN_BLOCKED);
+                    // }
+                    if self.consecutive_pin_mismatches >= 3 {
+                        return Err(Ctap2StatusCode::CTAP2_ERR_PIN_AUTH_BLOCKED);
+                    }
+                    let mut message = vec![0xFF; 32];
+                    message.extend(&[0x06, 0x08]);
+                    message.extend(&[min_pin_length as u8, 0x00, 0x00, 0x00]);
+                    // TODO(kaczmarczyck) commented code is useful for the extension
+                    // if !cbor::write(cbor_array_vec!(min_pin_length_rp_ids), &mut message) {
+                    //     return Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_RESPONSE_CANNOT_WRITE_CBOR);
+                    // }
+                    if !check_pin_auth(&self.pin_uv_auth_token, &message, &pin_auth) {
+                        return Err(Ctap2StatusCode::CTAP2_ERR_PIN_AUTH_INVALID);
+                    }
+                }
+                None => return Err(Ctap2StatusCode::CTAP2_ERR_PIN_AUTH_INVALID),
+            };
+        }
+        if min_pin_length < persistent_store.min_pin_length() {
+            return Err(Ctap2StatusCode::CTAP2_ERR_PIN_POLICY_VIOLATION);
+        }
+        persistent_store.set_min_pin_length(min_pin_length);
+        // if let Some(min_pin_length_rp_ids) = min_pin_length_rp_ids {
+        //     persistent_store.set_min_pin_length_rp_ids(min_pin_length_rp_ids)?;
+        // }
```

Comment:
Is this also a TODO?

Reply:
The above comment was meant to cover that, too. Copied for clarity.